status-im/status-react
# Wallet Connect transactions

fixes #20361
fixes #20805

### Summary

Finished the message signing user story, according to the [acceptance criteria](https://www.notion.so/I-want-to-sign-transactions-from-dApps-7681b4ddc5a94e8484e988445e6231b1?pvs=4). 

The PR includes the following changes:
* in the transaction modal, show the full testnet name (e.g. Mainnet -> Mainnet Sepolia)
* preparing/formatting/building the tx before showing the modal, so the user can:
  * see the exact transaction they will be signing
  * see the max fees based
* red highlight for the max fees when the user doesn't have enough balance to cover them
* changed the logic fo processing events to:
  * allow them to fail without showing the modal
  * show the request modal with the data already available, since `getSuggestedFees` and `buildTransaction` are async
* refactored the max fees estimation sub to account for the dynamic fees (EIP1559)
* removed support for eth_signTransaction


### Testing notes
[Instructions for connecting to dApps](https://www.notion.so/Testing-Wallet-Connect-interactions-0739f176856848999f0ea1671e5dfbe0)

### Steps to test


- Open Status
- Log in
- Make sure you are connected to a dApp _(check the **Testing notes** above)_
- From the dApp, trigger the following events 
  - `eth_signTransaction` (so far can only be tested on the test dapps)
  - `eth_sendTransaction` (the instructions include popular dapps where this can be done)
- Slide to authorise the transaction with password/biometric
- The dApp should update with a success state



### Risk

Described potential risks and worst case scenarios.

Tick **one**:

 
status: ready
